xma_data_processor.py

```diff
@@ -0,0 +1,374 @@
+            parts = [name.rsplit("_", 1)[0] for name in names]
+            if split_markers:
+                parts = parts + [f"sw_{part}" for part in parts]
+            if crossed_markers:
+                parts = parts + [
+                    f"cx_{part}_cam1x2"
+                    for part in [name.rsplit("_", 2)[0] for name in names]
+                ]
+        elif self._mode in ["2D", "per_cam"]:
+            parts = [name.rsplit("_", 2)[0] for name in names]
+        else:
+            raise SyntaxError("Invalid value for mode parameter")
+
+        parts_unique = []
+        for part in parts:
+            if part not in parts_unique:
+                parts_unique.append(part)
+        return parts_unique
+
+    def make_rgb_video(self, data_path):
+        """For all trials in given data path merges 2 videos into single RBG video."""
+        trials = [
+            folder
+            for folder in os.listdir(data_path)
+            if os.path.isdir(os.path.join(data_path, folder))
+            and not folder.startswith(".")
+        ]
+        for trial in trials:
+            path_to_trial = f"{data_path}/{trial}"
+            self._merge_rgb(path_to_trial)
+            substitute_data_relpath = "labeled-data/" + self._config["dataset_name"]
+            substitute_data_abspath = os.path.join(
+                os.path.split(self._config["path_config_file"])[0],
+                substitute_data_relpath,
+            )
+            self._extract_matched_frames_rgb(
+                path_to_trial,
+                substitute_data_abspath,
+                range(1, self._config["nframes"] + 1),
+            )
+            self._splice_xma_to_dlc(
+                path_to_trial,
+                swap=self._config["swapped_markers"],
+                cross=self._config["crossed_markers"],
+            )
+
+    def _merge_rgb(self, trial_path, codec="avc1", mode="difference"):
+        """
+        Takes the path to a trial subfolder and exports a single new video with
+        cam1 video written to the red channel and cam2 video written to the
+        green channel. The blue channel is, depending on the value of config
+        "mode", either the difference blend between A and B, the multiply
+        blend, or just a black frame.
+        """
+        trial_name = os.path.basename(os.path.normpath(trial_path))
+        if os.path.exists(f"{trial_path}/{trial_name}_rgb.avi"):
+            print("RGB video already created. Skipping.")
+            return
+        try:
+            cam1_video = cv2.VideoCapture(f"{trial_path}/{trial_name}_cam1.avi")
+        except FileNotFoundError as e:
+            raise FileNotFoundError(
+                f"Make sure your cam1 video for trial {trial_name} is named {trial_name}_cam1.avi"
+            ) from e
+        try:
+            cam2_video = cv2.VideoCapture(f"{trial_path}/{trial_name}_cam2.avi")
+        except FileNotFoundError as e:
+            raise FileNotFoundError(
```

Comment:
This can be simplified to "raise e"... I sound like a broken record at this point, but Python syntax (especially error handling) was very new to me when I first wrote this code a year ago.

Reply:
if we don't want custom error message we then don't need try/except/raise at all? I updated it to remove the extra code, lmk if i misunderstand